### DISCUSSION TITLE
Emulate linux inside the browser along with Jupyterlite

### DISCUSSION THREAD

**Discussion by thisistaimur** [jupyterlite/jupyterlite#1226] [Ideas]
Something like [v86](https://github.com/humphd/v86/tree/filer-9p-lastknowngood) or [Browser-Shell ](https://github.com/humphd/browser-shell) can be used to emulate linux in the browser, where other environments can be installed and emulated as well. 

This might be an amazing feature for handling small tasks, and for teaching/learning Jupyter. For example, teaching Jupyter on HPCs for example without needing an actual HPC.

**Comment by jtpio** [comment]
Yes that would be great indeed!

Those could probably be experimented with in as a third-party extension. Ideally it could also be integrated with the JupyterLab terminal:

![image](https://github.com/jupyterlite/jupyterlite/assets/591645/012b17dd-1677-4b2e-931e-7ecaf6b497bd)

**Reply by thisistaimur** [reply]
Yeah JupyterLab terminal is how I was imagining this too. If there are people who want to collaborate on this then feel free to reach to discuss possibilities!
